model-lowerings-common/src/org/jetbrains/dukat/model/commonLowerings/overrides/VarConflictResolveLowering.kt

```diff
@@ -0,0 +1,200 @@
+package org.jetbrains.dukat.model.commonLowerings.overrides
+
+import org.jetbrains.dukat.astCommon.IdentifierEntity
+import org.jetbrains.dukat.astCommon.NameEntity
+import org.jetbrains.dukat.astCommon.appendLeft
+import org.jetbrains.dukat.astCommon.shiftRight
+import org.jetbrains.dukat.astModel.ClassLikeModel
+import org.jetbrains.dukat.astModel.ClassModel
+import org.jetbrains.dukat.astModel.InterfaceModel
+import org.jetbrains.dukat.astModel.MemberModel
+import org.jetbrains.dukat.astModel.ModuleModel
+import org.jetbrains.dukat.astModel.PropertyModel
+import org.jetbrains.dukat.astModel.SourceSetModel
+import org.jetbrains.dukat.astModel.TypeModel
+import org.jetbrains.dukat.model.commonLowerings.MemberData
+import org.jetbrains.dukat.model.commonLowerings.ModelContext
+import org.jetbrains.dukat.model.commonLowerings.ModelLowering
+import org.jetbrains.dukat.model.commonLowerings.ModelWithOwnerTypeLowering
+import org.jetbrains.dukat.model.commonLowerings.OverrideTypeChecker
+import org.jetbrains.dukat.model.commonLowerings.allParentMembers
+import org.jetbrains.dukat.model.commonLowerings.buildInheritanceGraph
+import org.jetbrains.dukat.ownerContext.NodeOwner
+
+private fun PropertyModel.toVal(): PropertyModel {
+    return copy(
+        immutable = true,
+        setter = false
+    )
+}
+
+private class VarOverrideResolver(
+    private val modelContext: ModelContext,
+    private val inheritanceContext: InheritanceContext
+) : ModelWithOwnerTypeLowering {
+
+    val propertiesToChangeToVal: MutableSet<NameEntity> = mutableSetOf()
+
+    private fun findConflictingProperties(
+        parentMembers: Map<NameEntity?, List<MemberData>>,
+        typeChecker: OverrideTypeChecker
+    ): List<List<MemberData>> {
+        return parentMembers.keys.mapNotNull { name ->
+            val properties = parentMembers[name].orEmpty().filter { it.memberModel is PropertyModel }
+
+            val propertiesInImmediateParents = properties.filter { first ->
+                properties.none { second ->
+                    inheritanceContext.isDescendant(second.ownerModel, first.ownerModel)
```

Comment:
We should collect actual parent instead collecting all parents and then filtering out this way ) (check out ModelContext::getParents)

Reply:
What about this case?
```
interface A {
  x: number
}

interface B : A {
}

interface C : B {
  x: number
}
```
I still want to get x from A in this case, that's why I can't use something like ModelContext::getParents. So I believe I'm right in this place -- the name can be misleading though, can you suggest what change it to?